### PR TITLE
Allow resolution to be a variable in an alert

### PR DESCRIPTION
**Pull request by dmichel1**
Also check for variables in aggregations.

### CODE REVIEW

**Comment by ryanvazquez** [comment]
LGTM

File: src/datasource.ts [hunk 1]
```diff
@@ -227,14 +227,18 @@ export default class HeroicDatasource {
...
+      for (const aggregation of select) {
+        for (const param of aggregation.params) {
+          if (this.templateSrv.variableExists(param)) {
+            return true;
+          }
         }
       }
     }
 
+    // Check filters for variables.
```
Comment (lmuhlha) [inline src/datasource.ts]: wait so this is supposed to work? re: can't filter w/ variables

Comment (dmichel1) [reply src/datasource.ts]: @lmuhlha this function `targetContainsTemplate` get's called from the grafana alert panel. what this function does now is verify aggregations and filters **do not** contain variables when the query is used in an alert. If they do a warning is shown.

Comment (lmuhlha) [reply src/datasource.ts]: ah gotcha 👍

Comment (lmuhlha) [reply src/datasource.ts]: maybe a more useful comment would be "ensure filters do not contain variables" but 🤷‍♀

**Pull request merged**